core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -3710,6 +3711,33 @@ protected void addInitialAdminData(boolean addAdmin, boolean initialSetup) throw
                     log.error(message);
                 }
             } else if (addAdmin) {
+                if (!checkUserNameValid(adminUserName)) {
```

Comment:
I suppose this validation is same as the [1]. Rather duplicating the code, can you extract that logic to a private method?

[1] https://github.com/dracusds123/carbon4-kernel/blob/IDENTITY-3690/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java#L1371